PyPoE/cli/exporter/wiki/parsers/lua.py

```diff
@@ -506,7 +506,7 @@ def main(self, parsed_args):
             self._copy_from_keys(
                 row, self._COPY_KEYS_BESTIARY_COMPONENTS, components
             )
-            if row['BeastRarity'] != RARITY.ANY:
+            if row['BeastRarity'] != RARITY.ANY and not (row['BeastRarity'] is None):
```

Comment:
```py
if row['BeastRarity'] != RARITY.ANY and row['BeastRarity'] is not None:
  ...
```
Convention-wise it is always `x is not None` in conditions. In general, as a tip, I would recommend reading the condition in your head to see if it makes sense as a sentence. In this case - `"beast rarity is not any and not beast rarity is none."` as a sentence does not work, whereas `"beast rarity is not any and beast rarity is not none"` sounds a lot better. 

Additionally, in this case there's no need to wrap what comes after `and` in parentheses. The interpreter understands the condition. Normally you'd employ parentheses for improved readability or to change the order of the ops, but in this case it is not necessary as the order is semantic: `if beast rarity isn't ANY and it isn't None then..` which removes the need for parentheses.

Reply:
Hunh. For some reason I had it stuck in my head that `x is not None` didn't work, and I had to do `not x is None`.
I agree, I much prefer `x is not None`.

The parens were added to help readability of the statement prefixed by `not`. I'm aware they don't need to wrap the condition following the `and`.
Thanks for all the explanation though. You don't know what I do and don't know :)